### PR TITLE
refactor R-CNN for autogluon. 

### PR DESCRIPTION
**Pull request by Jerryzcn**
MXNet Oct16 works. MXNet version later than Oct16 will encounter OOM or index error @chongruo

### CODE REVIEW

**Comment by zhanghang1989** [comment]
@chongruo

File: gluoncv/model_zoo/rcnn/faster_rcnn/faster_rcnn.py [hunk 1]
```diff
@@ -491,7 +502,7 @@ def get_faster_rcnn(name, dataset, pretrained=False, ctx=mx.cpu(),
     """
     net = FasterRCNN(**kwargs)
     if pretrained:
-        from ..model_store import get_model_file
+        from gluoncv.model_zoo.model_store import get_model_file
```
Comment (zhreshold) [inline gluoncv/model_zoo/rcnn/faster_rcnn/faster_rcnn.py]: why not use relative import?

File: gluoncv/model_zoo/rcnn/faster_rcnn/faster_rcnn.py [hunk 2]
```diff
@@ -756,7 +767,7 @@ def faster_rcnn_resnet50_v1b_custom(classes, transfer=None, pretrained_base=True
     if pretrained:
         warnings.warn("Custom models don't provide `pretrained` weights, ignored.")
     if transfer is None:
-        from ..resnetv1b import resnet50_v1b
+        from gluoncv.model_zoo.resnetv1b import resnet50_v1b
```
Comment (zhreshold) [inline gluoncv/model_zoo/rcnn/faster_rcnn/faster_rcnn.py]: ...model_zoo

File: gluoncv/model_zoo/rcnn/faster_rcnn/faster_rcnn.py [hunk 3]
```diff
@@ -969,7 +1037,7 @@ def faster_rcnn_resnet101_v1d_custom(classes, transfer=None, pretrained_base=Tru
     if pretrained:
         warnings.warn("Custom models don't provide `pretrained` weights, ignored.")
     if transfer is None:
-        from ..resnetv1b import resnet101_v1d
+        from gluoncv.model_zoo.resnetv1b import resnet101_v1d
```
Comment (zhreshold) [inline gluoncv/model_zoo/rcnn/faster_rcnn/faster_rcnn.py]: ...model_zoo

File: gluoncv/model_zoo/rcnn/faster_rcnn/faster_rcnn.py [hunk 4]
```diff
@@ -983,18 +1051,86 @@ def faster_rcnn_resnet101_v1d_custom(classes, transfer=None, pretrained_base=Tru
...
-            rpn_channel=1024, base_size=16, scales=(2, 4, 8, 16, 32),
-            ratios=(0.5, 1, 2), alloc_size=(128, 128), rpn_nms_thresh=0.7,
-            rpn_train_pre_nms=12000, rpn_train_post_nms=2000,
-            rpn_test_pre_nms=6000, rpn_test_post_nms=300, rpn_min_size=16,
-            num_sample=128, pos_iou_thresh=0.5, pos_ratio=0.25, max_num_gt=300,
-            **kwargs)
+            train_patterns=train_patterns, **kwargs)
     else:
-        from ...model_zoo import get_model
+        from gluoncv.model_zoo import get_model
```
Comment (zhreshold) [inline gluoncv/model_zoo/rcnn/faster_rcnn/faster_rcnn.py]: ...model_zoo

File: gluoncv/model_zoo/rcnn/faster_rcnn/faster_rcnn.py [hunk 5]
```diff
@@ -983,18 +1051,86 @@ def faster_rcnn_resnet101_v1d_custom(classes, transfer=None, pretrained_base=Tru
...
+        features, top_features, box_features = \
+            custom_rcnn_fpn(pretrained_base, base_network_name, norm_layer, norm_kwargs,
+                            sym_norm_layer, sym_norm_kwargs, num_fpn_filters, num_box_head_conv,
+                            num_box_head_conv_filters, num_box_head_dense_filters)
+        return get_faster_rcnn(
+            name='fpn_' + base_network_name, dataset=dataset, pretrained=False, features=features,
+            top_features=top_features, classes=classes, box_features=box_features,
+            train_patterns=train_patterns, **kwargs)
+    else:
+        from gluoncv.model_zoo import get_model
```
Comment (zhreshold) [inline gluoncv/model_zoo/rcnn/faster_rcnn/faster_rcnn.py]: same here

File: gluoncv/model_zoo/rcnn/mask_rcnn/mask_rcnn.py [hunk 6]
```diff
@@ -360,7 +362,7 @@ def get_mask_rcnn(name, dataset, pretrained=False, ctx=mx.cpu(),
     """
     net = MaskRCNN(**kwargs)
     if pretrained:
-        from ..model_store import get_model_file
+        from gluoncv.model_zoo.model_store import get_model_file
```
Comment (zhreshold) [inline gluoncv/model_zoo/rcnn/mask_rcnn/mask_rcnn.py]: ...model_zoo

**Comment by mli** [comment]
Job PR-1166-8 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1166/8/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1166-9 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1166/9/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1166-10 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1166/10/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1166-11 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1166/11/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1166-12 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1166/12/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1166-13 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1166/13/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1166-22 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1166/22/index.html 
Code coverage of this PR:  vs. Master:

File: gluoncv/model_zoo/rcnn/rpn/rpn_target.py [hunk 7]
```diff
@@ -5,11 +5,11 @@
...
 from mxnet import autograd, gluon
 
-from ...nn.bbox import BBoxSplit
-from ...nn.coder import SigmoidClassEncoder, NumPyNormalizedBoxCenterEncoder
+from ....nn.bbox import BBoxSplit
+from ....nn.coder import SigmoidClassEncoder, NumPyNormalizedBoxCenterEncoder
 
 try:
-    import gluoncv.model_zoo.rpn.cython_rpn_target as cython_rpn_target
+    import gluoncv.model_zoo.rcnn.rpn.cython_rpn_target as cython_rpn_target
```
Comment (zhreshold) [inline gluoncv/model_zoo/rcnn/rpn/rpn_target.py]: I  don't know why it was importing from gluoncv, but should use relative import here again

**Comment by mli** [comment]
Job PR-1166-24 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1166/24/index.html 
Code coverage of this PR:  vs. Master:

File: scripts/detection/faster_rcnn/train_faster_rcnn.py [hunk 8]
```diff
@@ -32,6 +32,7 @@
 from gluoncv.utils.parallel import Parallelizable, Parallel
 from gluoncv.utils.metrics.rcnn import RPNAccMetric, RPNL1LossMetric, RCNNAccMetric, \
     RCNNL1LossMetric
+from gluoncv.data import COCODetection, VOCDetection
```
Comment (zhreshold) [inline scripts/detection/faster_rcnn/train_faster_rcnn.py]: version check should be 0.7.0

File: scripts/instance/mask_rcnn/train_mask_rcnn.py [hunk 9]
```diff
@@ -31,6 +31,7 @@
 from gluoncv.utils.metrics.rcnn import RPNAccMetric, RPNL1LossMetric, RCNNAccMetric, \
     RCNNL1LossMetric, MaskAccMetric, MaskFGAccMetric
 from gluoncv.utils.parallel import Parallelizable, Parallel
+from gluoncv.data import COCODetection, VOCDetection
```
Comment (zhreshold) [inline scripts/instance/mask_rcnn/train_mask_rcnn.py]: version check 0.7.0

**Comment by mli** [comment]
Job PR-1166-25 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1166/25/index.html 
Code coverage of this PR:  vs. Master:

**Pull request merged**